### PR TITLE
SUBMARINE-560. Build an API to manage notebook instances with Swagger

### PR DESCRIPTION
**Pull request by lowc1012**
### What is this PR for?
User can create his or her notebook instances in specified namespace.

### What type of PR is it?
[Feature]

### Todos
Will open another PR to finish following todos.
1.  Add test for Notebook Rest API
2. User can list/delete notebook instances via REST API

### What is the Jira issue?
[SUBMARINE-560](https://issues.apache.org/jira/projects/SUBMARINE/issues/SUBMARINE-560)

### How should this be tested?
[Travis CI](https://travis-ci.org/github/lowc1012/submarine/builds/711003012)

### Screenshots (if appropriate)
image1

image2

### Questions:
* Does the licenses files need update? No
* Is there breaking changes for older versions? No
* Does this needs documentation? No

### CODE REVIEW

**Comment by lowc1012** [comment]
There are some problem with notebook pod.
Once initContainer executed completely, the pod's status turned to CrashLoopBackOff.
WIP.

**Comment by liuxunorg** [comment]
@lowc1012 Please fix Conflicting files problem.

File: dev-support/docker-images/jupyter/build.sh [hunk 1]
```diff
@@ -16,7 +16,7 @@
 
 set -euxo pipefail
 
-TF_JUPYTER_IMAGE="apache/submarine:tf2.1.0-jupyter"
+JUPYTER_IMAGE="apache/submarine:jupyter-notebook"
```
Comment (liuxunorg) [inline dev-support/docker-images/jupyter/build.sh:19]: submarine docker image name format is `apache/submarine:{component-name}-{version}`, like., `apache/submarine:mini-0.5.0-SNAPSHOT`
please view : https://hub.docker.com/r/apache/submarine/tags
So I think the `JUPYTER_IMAGE` name should be `apache/submarine:jupyter-notebook-0.5.0-SNAPSHOT`

@pingsutw can you help @lowc1012 push `JUPYTER_IMAGE` push to docker hub repo?

**Comment by lowc1012** [comment]
@liuxunorg  
Pushed a fixed-commit.
Thanks for your review!

File: dev-support/docker-images/jupyter/build.sh (hunk 1 above)
Comment (pingsutw) [reply dev-support/docker-images/jupyter/build.sh]: I just pushed the image to https://hub.docker.com/r/apache/submarine/tags

**Comment by lowc1012** [comment]
Hi @jiwq,
It can be deployed by [submarine helm charts](https://github.com/apache/submarine/tree/master/helm-charts/submarine/charts/notebook-controller)
Sorry. I should update the [docs](https://github.com/apache/submarine/blob/master/docs/userdocs/k8s/notebook.md) for detailed information.

**Comment by jiwq** [comment]
> Hi @jiwq,
> It can be deployed by [submarine helm charts](https://github.com/apache/submarine/tree/master/helm-charts/submarine/charts/notebook-controller), or kubectl apply -k submarine/dev-support/k8s/notebook-controller
> Sorry. I should update the [docs](https://github.com/apache/submarine/blob/master/docs/userdocs/k8s/notebook.md) for detailed information.

+1 Thanks for your help.  @lowc1012

**Comment by liuxunorg** [comment]
@lowc1012 Thank you contribution this important feature.
Will merge if no more comments.

**Pull request closed without merging**